File: 1° semestre/APC verão/Raylan da Silva Sales_lab_4/C/Lab4a.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

int binsearch(int v[], int p, int i, int f);

int main(){
	int *v, p, aux;
	int qtd, i, pv, j, q;

	printf("Digite o tamanho do vetor: ");
	scanf("%d", &qtd);

	v = (int*) malloc(qtd*sizeof(int));
	q = qtd;

	for(i=0;i<qtd;i++){
        printf("Digite %d numeros: ", q);
		scanf("%d", &v[i]);
		q--;
	}

	for(i=0;i<qtd;i++){
		for(j=0;j<qtd-1;j++){
			if(v[j]>v[j+1]){
				aux = v[j];
				v[j] = v[j+1];
				v[j+1] = aux;
			}
		}
	}

	printf("Qual numero voce gostaria de procurar? ");
	scanf("%d", &p);
	pv = binsearch( v, p, 0, qtd-1);

	printf("A posicao do valor %d no vetor e %d", p, pv);

	return 0;
}

int binsearch(int v[], int p, int i, int f){
	int m;

	if(i>f)
		return -1;

	m = (f+i)/2;

	if(v[m]==p)
		return m;

	else if(v[m]>p)
		return binsearch( v, p, i, m-1);

	else
		return binsearch( v, p, m+1, f);

}

File: 1° semestre/APC verão/Raylan da Silva Sales_lab_4/Python/Lab4a.py
def binary_search_rec(vet, num, esq, dir):
	meio = (esq + dir) // 2
	aux_num = vet[meio]
	if num == aux_num:
		return meio
	elif num > aux_num:
		return binary_search_rec(vet, num, meio-1, dir)
	return binary_search_rec(vet, num, esq, meio+1,)

def bubble_sort(lista):
    elementos = len(lista)-1
    ordenado = False
    while not ordenado:
        ordenado = True
        for i in range(elementos):
            if lista[i] > lista[i+1]:
                lista[i], lista[i+1] = lista[i+1],lista[i]
                ordenado = False
    return lista

vetor = [16]
cont = 0
while cont<15:
    entrada = int(input("Digite a entrada: "))
    vetor.insert(cont, entrada)
    cont+=1

vetor = bubble_sort(vetor)

busca = int(input("Qual numero voce quer procurar? "))

pv = binary_search_rec(vetor, busca, 0, 15)

print("A posicao do valor {} no vetor eh: {}".format(busca, pv))

